packages/dm-core-plugins/src/form/fields/NumberField.tsx

```diff
@@ -47,19 +47,24 @@ export const NumberField = (props: TNumberFieldProps) => {
         required: !optional,
         pattern: { value: /^[0-9]+$/g, message: 'Only digits allowed' },
       }}
-      defaultValue={defaultValue || undefined}
+      defaultValue={defaultValue || null}
```

Comment:
react-hook-form does not allow undefined as a value
https://react-hook-form.com/api/usecontroller/controller/